magic_items/routes.py

```diff
@@ -0,0 +1,304 @@
+            flash("new item " + edited_item.name + " edited", "success")
+            return redirect(f"/magic-items/{edited_item.id}")
+
+        except:
+            flash("error", "danger")
+            return render_template(
+                "magic_items/edit_magic_item.html", form=form, item=item
+            )
+
+    return render_template("magic_items/edit_magic_item.html", form=form, item=item)
+
+
+@magic_routes.route("/magic-items/delete", methods=["POST"])
+@before_manage_magic_items
+def delete_item():
+
+    magic_item_id = int(request.args["magic_item_id"])
+    item = MagicItemService.get(magic_item_id)
+
+    if g.user.id is not item.created_by:
+        flash("Not authorized to make changes to that magic item", "danger")
+        return redirect(f"/magic-items?created_by={g.user.id}")
+
+    MagicItemService.delete(g.user.id, magic_item_id)
+    flash(f"{item.name} has been deleted", "success")
+    return redirect(f"/magic-items?created_by={g.user.id}")
+
+
+@magic_routes.route("/magic-items/random", methods=["GET"])
+def random_item():
+    item = MagicItemService.random()
+    return render_template("magic_items/magic_item.html", item=item)
+
+
+#########################################################################################################
+#########################################################################################################
+##
+## Collection Routes
+
+
+@magic_routes.route("/collections/new", methods=["GET", "POST"])
+@before_manage_collections
+def add_new_collection():
+
+    form = CollectionAddForm()
+
+    if form.validate_on_submit():
+        if not request.args:
+            try:
+                collection = CollectionService.create(g.user.id, form)
+                flash("new collection " + collection.name + " added", "success")
+                return redirect(f"/collections/{collection.id}")
+
+            except:
+                flash("error", "danger")
+                return render_template("magic_items/new_collection.html", form=form)
+        else:
+            try:
+                magic_item_id = int(request.args["magic_item_id"])
+                collection = CollectionService.create(g.user.id, form)
+                CollectionService.add_magic_item(magic_item_id, collection.id)
+                flash(f"new collection {collection.name} created", "success")
+                return redirect(f"/collections/{collection.id}")
+
+            except:
+                flash("error", "danger")
+                return render_template("magic_items/new_collection.html", form=form)
+
+    return render_template("magic_items/new_collection.html", form=form)
+
+
+@magic_routes.route("/collections", methods=["GET"])
+def show_collections():
+
+    if request.args and request.args["user_id"] == "":
+        flash("Please login or signup to manage your collections", "danger")
+        return redirect("/login")
```

Comment:
We can't use g.user here? Why do we expect the user_id being passed via request.args?